src/main/java/edu/harvard/iq/dataverse/DatasetVersionServiceBean.java

```diff
@@ -154,6 +154,26 @@ public DatasetVersion getDatasetVersion(){
     public DatasetVersion find(Object pk) {
         return em.find(DatasetVersion.class, pk);
     }
+    
+    public DatasetVersion findDeep(Object pk) {
+        return (DatasetVersion) em.createNamedQuery("DatasetVersion.findById")
+            .setParameter("id", pk)
+            // Optimization hints: retrieve all data in one query; this prevents point queries when iterating over the files 
+            .setHint("eclipselink.left-join-fetch", "o.fileMetadatas.dataFile.ingestRequest")
+            .setHint("eclipselink.left-join-fetch", "o.fileMetadatas.dataFile.thumbnailForDataset")
+            .setHint("eclipselink.left-join-fetch", "o.fileMetadatas.dataFile.dataTables")
```

Comment:
This is a very good point. I just haven't gotten around to finding a dataset with a ton of tabular files to confirm that this is indeed going to result in observable savings... But I can't think of a good reason not to drop this hint, no. 
(will find a test dataset now)

Reply:
And yes, I get the part that it's not just the DataTables.

Reply:
So, does this really mean that we don't need the hint for roleassignments either? Or rather, that the only hints we need are for the 1:1 relations on DataFile? (that would otherwise result in additional individual queries?)

Reply:
(The awful loop above can be fixed of course; it needs to be fixed. I was just demoralized a little by seeing things like that. The above happens 40 lines into the init method. There's almost certainly more of the same throughout the rest of the page. Things that were already in 5.13, that had gradually crept into the code, that were making things slow on some large datasets in 5.13 and are offsetting the gains from the 5.14 optimizations now. I wasn't mentally prepared for going this deep. But will keep looking into it.)

Reply:
Ah fun - needed for the download all button I guess. This type of thing definitely looks like a great little post 5.14 spin-out issue (that the SPA will also have).

Reply:
Hmm. I became convinced last night that this really needs to be addressed as part of this PR. At least some of it, the more glaring stuff. The "awful loop" above - it is awful because it goes through every file in the entire dataset; negating the benefits of trying to only initialize one version-worth of files. AND it doesn't appear to be necessary - because for all the useful purposes the page and its fragment only need to know if tab files are present in the current version. (There is a separate loop for the tab files in the version there). 
Dropping the hint for DataTable for the version doesn't appear to be as straightforward through - the page in its current form needs all the DataTables in the version, both for checking if there are tab files, and for calculating the "original" sizes, if tab files are present. Can (and probably should) be replaced with custom queries, but this may be an effort that can be delayed - will see how it goes.

Reply:
... the full sentence should have been "negating the benefits of trying to only initialize one version-worth of files *in some large datasets with sparseley-populated versions*" of course.